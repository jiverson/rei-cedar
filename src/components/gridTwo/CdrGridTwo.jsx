import style from './styles/CdrGridTwo.scss';

export default {
  name: 'CdrGridTwo',
  data() {
    return {
      style,
    };
  },
  render() {
    return (
      <div class={this.style['cdr-grid-two']}>
        {this.$slots.default}
      </div>
    );
  },
};
