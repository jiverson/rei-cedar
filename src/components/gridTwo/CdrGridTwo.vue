<template>
  <component
    :is="tag"
    :class="componentClass">
    <slot />
  </component>
</template>
<script lang="ts">
import { defineComponent, computed } from 'vue';
import clsx from 'clsx';
import { buildClass } from '../../utils/buildClass'

import propValidator from '../../utils/propValidator';
import style from './styles/CdrGridTwo.scss';

export default {
  name: 'CdrGridTwo',
  props: {
    /**
     * Defines gutter size.
     * Possible values: {none, small, medium, large}.
     * Also accepts responsive values with `@breakpoint`: "none@md"
     */
    gutter: {
      type: String,
      validator: (value) => propValidator(
        value,
        ['none', 'small', 'medium', 'large'],
      ),
      default: 'medium@xs medium@sm large@md large@lg',
    },
    tag: {
      type: String,
      default: 'div',
    },
  },
  setup(props) {
    // const Component = this.tag;
    // {clsx(this.style['cdr-grid-two'], this.gutterClass)}





    gutterClass() {
      const classStr = [];
      if (this.gutter) {
        this.gutter.split(' ').forEach((val) => {
          classStr.push(this.style[`cdr-grid-two--gutter-${val}`]);
        });
      }
      return classStr.join(' ');
    },
    return (

    );
  },
};

</script>
