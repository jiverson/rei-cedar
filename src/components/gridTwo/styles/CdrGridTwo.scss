@import '../../../css/settings/index.scss';
@import './vars/CdrGridTwo.vars.scss';

.cdr-grid-two {
  @include cdr-grid-two-base-mixin;

  // Responsive Gutters
  &--gutter-none {
    @include cdr-grid-two-gutter-none-mixin;
  }

  &--gutter-small {
    @include cdr-grid-two-gutter-small-mixin;
  }

  &--gutter-medium {
    @include cdr-grid-two-gutter-medium-mixin;
  }

  &--gutter-large {
    @include cdr-grid-two-gutter-large-mixin;
  }

  @include cdr-xs-mq-only {
    &--gutter-none\@xs {
      @include cdr-grid-two-gutter-none-mixin;
    }

    &--gutter-small\@xs {
      @include cdr-grid-two-gutter-small-mixin;
    }

    &--gutter-medium\@xs {
      @include cdr-grid-two-gutter-medium-mixin;
    }

    &--gutter-large\@xs {
      @include cdr-grid-two-gutter-large-mixin;
    }
  }
  @include cdr-sm-mq-only {
    &--gutter-none\@sm {
      @include cdr-grid-two-gutter-none-mixin;
    }

    &--gutter-small\@sm {
      @include cdr-grid-two-gutter-small-mixin;
    }

    &--gutter-medium\@sm {
      @include cdr-grid-two-gutter-medium-mixin;
    }

    &--gutter-large\@sm {
      @include cdr-grid-two-gutter-large-mixin;
    }
  }
  @include cdr-md-mq-only {
    &--gutter-none\@md {
      @include cdr-grid-two-gutter-none-mixin;
    }

    &--gutter-small\@md {
      @include cdr-grid-two-gutter-small-mixin;
    }

    &--gutter-medium\@md {
      @include cdr-grid-two-gutter-medium-mixin;
    }

    &--gutter-large\@md {
      @include cdr-grid-two-gutter-large-mixin;
    }
  }
  @include cdr-lg-mq-only {
    &--gutter-none\@lg {
      @include cdr-grid-two-gutter-none-mixin;
    }

    &--gutter-small\@lg {
      @include cdr-grid-two-gutter-small-mixin;
    }

    &--gutter-medium\@lg {
      @include cdr-grid-two-gutter-medium-mixin;
    }

    &--gutter-large\@lg {
      @include cdr-grid-two-gutter-large-mixin;
    }
  }
}
