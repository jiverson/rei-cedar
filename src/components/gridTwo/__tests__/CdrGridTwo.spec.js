import { shallowMount } from '../../../../test/vue-jest-style-workaround.js';
import CdrGridTwo from 'componentdir/gridTwo/CdrGridTwo';

describe('CdrGridTwo', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(CdrGridTwo);
    expect(wrapper.element).toMatchSnapshot();
  });


  it('complex example matches snapshot', () => {
    const wrapper = shallowMount(CdrGridTwo, {
      propsData: {
        gutter: 'none@xs large@sm medium@md small@sm',
        tag: 'ul'
      },
      slots: {
        default: '<li>hey</li>'
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
