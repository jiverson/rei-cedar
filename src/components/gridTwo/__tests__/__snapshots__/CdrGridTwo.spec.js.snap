// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CdrGridTwo complex example matches snapshot 1`] = `
<ul
  class="cdr-grid-two cdr-grid-two--gutter-none@xs cdr-grid-two--gutter-large@sm cdr-grid-two--gutter-medium@md cdr-grid-two--gutter-small@sm"
>
  <li>
    hey
  </li>
</ul>
`;

exports[`CdrGridTwo matches snapshot 1`] = `
<div
  class="cdr-grid-two cdr-grid-two--gutter-medium@xs cdr-grid-two--gutter-medium@sm cdr-grid-two--gutter-large@md cdr-grid-two--gutter-large@lg"
/>
`;
